frontend/src/metabase/components/ErrorPages/ErrorDiagnosticModal.tsx

```diff
@@ -90,58 +190,141 @@ export const ErrorDiagnosticModal = ({
+                    {t`This information helps ${applicationName} figure out what exactly caused the issue.`}
+                  </Text>
+                </Stack>
+                {enableBugReportField && (
+                  <Button
+                    variant="subtle"
+                    p="0"
+                    style={{ fontSize: "17px", width: "50px" }}
+                    onClick={() => setIsDetailsVisible(!isDetailsVisible)}
+                  >
+                    {isDetailsVisible ? t`Done` : t`Edit`}
+                  </Button>
+                )}
+              </Flex>
+              {(isDetailsVisible || !enableBugReportField) && (
+                <Stack spacing="md" pt="md">
+                  {canIncludeQueryData && (
+                    <FormCheckbox
+                      name="queryResults"
+                      label={t`Query results`}
+                    />
+                  )}
+                  {!!errorInfo.localizedEntityName && (
+                    <FormCheckbox
+                      name="entityInfo"
+                      label={t`${capitalize(
+                        errorInfo.localizedEntityName,
+                      )} definition`}
+                    />
+                  )}
+                  <FormCheckbox
+                    name="frontendErrors"
+                    label={t`Browser error messages`}
+                  />
+                  {!!errorInfo?.logs && (
+                    <>
+                      <FormCheckbox
+                        name="backendErrors"
+                        label={t`All server error messages`}
+                      />
+                      <FormCheckbox name="logs" label={t`All server logs`} />
+                      <FormCheckbox
+                        name="userLogs"
+                        label={t`Server logs from the current user only`}
+                      />
+                    </>
+                  )}
+                  <FormCheckbox
+                    name="bugReportDetails"
+                    label={t`${applicationName} instance version information`}
+                  />
+                </Stack>
+              )}
+            </Box>
+            {enableBugReportField ? (
               <>
-                <FormCheckbox
-                  name="backendErrors"
-                  label={t`All server error messages`}
-                />
-                <FormCheckbox name="logs" label={t`All server logs`} />
-                <FormCheckbox
-                  name="userLogs"
-                  label={t`Server logs from the current user only`}
+                <Button
+                  variant="filled"
+                  leftIcon={isSlackSending ? <Loader size="xs" /> : null}
+                  onClick={() => {
+                    trackSimpleEvent({
+                      event: "error_diagnostic_modal_submitted",
+                      event_detail: "submit-report",
+                    });
```

Comment:
check out existing usage of `trackSimpleEvent` - we always create one or more wrapper functions for analytics tracking and put them in a separate `analytics.ts` file to keep analytics easy to find and update.

Reply:
I noticed that, but wasn’t sure where would be the best place to put the analytics file. two tracking events are inside the modal, but there are 3 more (4 in near future) that are all over the place. do you think it would make more sense if e.g. opening bug report from command palette should reference analytics file buried in `ErrorPages` folder?

Reply:
You make a good point, It might be time for ErrorDiagnostics to move out and get its own folder